core/src/main/java/eu/kanade/tachiyomi/util/storage/SevenZUtil.kt

```diff
@@ -0,0 +1,23 @@
+package eu.kanade.tachiyomi.util.storage
+
+import eu.kanade.tachiyomi.util.lang.compareToCaseInsensitiveNaturalOrder
+import org.apache.commons.compress.archivers.dump.UnsupportedCompressionAlgorithmException
+import org.apache.commons.compress.archivers.sevenz.SevenZFile
+import tachiyomi.core.util.system.ImageUtil
+import java.io.IOException
+import java.io.InputStream
+
+object SevenZUtil {
+    fun SevenZFile.getImages(): Sequence<ByteArray> {
+        return generateSequence {
+            try {
+                getNextEntry()
+            } catch (e: IOException) {
+                throw UnsupportedCompressionAlgorithmException()
```

Comment:
Can you check if [this](https://commons.apache.org/proper/commons-compress/apidocs/org/apache/commons/compress/archivers/sevenz/SevenZFile.html#matches(byte%5B%5D,int)) helps?

Reply:
didn't notice that, thanks.

Reply:
i don't think it does what i need :thinking: